### PR TITLE
adding l4proto in flowrecord

### PR DESCRIPTION
**Pull request by 0xmchadha**
Report `@l4proto` in flow records.

> Part of project [protocol](https://github.com/orgs/aporeto-inc/projects/59)

### CODE REVIEW

**Comment by t00f** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "component": "jenova",
    "pr-id": "8",
    "commit-sha": "ce0b235cc0624ea2a21b141db2ca26852275f62e"
  },
  {
    "component": "gaia",
    "pr-id": "121",
    "commit-sha": "2e3c03df94394278a7fcf4b24b6f512c2caad0b1"
  },
  {
    "component": "clad",
    "pr-id": "549",
    "commit-sha": "2772718cbf4e3bb3a16ac7f394ac4d84aff64108"
  },
  {
    "component": "trireme-lib",
    "pr-id": "499",
    "commit-sha": "c2652d9c16de82b7c2a46ee742c177ab4650ac34"
  },
  {
    "component": "enforcerd",
    "pr-id": "532",
    "commit-sha": "5d478f5f931c6aab3b516f390520c29185746196"
  }
]
```

**Comment by t00f** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "component": "gaia",
    "pr-id": "121",
    "commit-sha": "21a2cdba6a6d448d3e96c6bd1f5db3baa2f89373"
  },
  {
    "component": "clad",
    "pr-id": "549",
    "commit-sha": "2772718cbf4e3bb3a16ac7f394ac4d84aff64108"
  },
  {
    "component": "trireme-lib",
    "pr-id": "499",
    "commit-sha": "c2652d9c16de82b7c2a46ee742c177ab4650ac34"
  },
  {
    "component": "enforcerd",
    "pr-id": "532",
    "commit-sha": "5d478f5f931c6aab3b516f390520c29185746196"
  },
  {
    "component": "jenova",
    "pr-id": "8",
    "commit-sha": "b4266a8fb19281a4f274af01d955d580864df46f"
  }
]
```

File: collector/interfaces.go [hunk 1]
```diff
@@ -100,15 +100,16 @@ type EndPoint struct {
...
 	Tags             *policy.TagStore
-	Action           policy.ActionType
-	ObservedAction   policy.ActionType
 	DropReason       string
 	PolicyID         string
 	ObservedPolicyID string
+	Count            int
+	Action           policy.ActionType
+	ObservedAction   policy.ActionType
+	L4Proto          uint8
```
Comment (t00f) [inline collector/interfaces.go]: Can we rename internal variables with complete names to make it clear ?
Use `L4Protocol` instead of `L4Proto`

**Review by t00f — CHANGES_REQUESTED** [review]

**Review by t00f — APPROVED** [review]

File: controller/internal/enforcer/applicationproxy/tcp/tcp.go [hunk 2]
```diff
@@ -22,6 +22,7 @@ import (
 	"github.com/aporeto-inc/trireme-lib/controller/internal/enforcer/constants"
 	"github.com/aporeto-inc/trireme-lib/controller/internal/enforcer/nfqdatapath/tokenaccessor"
 	"github.com/aporeto-inc/trireme-lib/controller/pkg/connection"
+	Packet "github.com/aporeto-inc/trireme-lib/controller/pkg/packet"
```
Comment (dstiliadis) [inline controller/internal/enforcer/applicationproxy/tcp/tcp.go]: No capital names in package names.

**Comment by t00f** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "component": "gaia",
    "pr-id": "121",
    "commit-sha": "c38b8870d6afbd3cefb13766bbb89a0b2ccfad30"
  },
  {
    "component": "clad",
    "pr-id": "549",
    "commit-sha": "2772718cbf4e3bb3a16ac7f394ac4d84aff64108"
  },
  {
    "component": "trireme-lib",
    "pr-id": "499",
    "commit-sha": "e6680c0951b75468fb85ccd7922cada078283228"
  },
  {
    "component": "enforcerd",
    "pr-id": "532",
    "commit-sha": "9e33f845175c7f11bce9ff0a9f2ca483761bcb55"
  },
  {
    "component": "jenova",
    "pr-id": "8",
    "commit-sha": "a71cca0c0bfec99dfeeed4d24b30a187d59f60b6"
  }
]
```

**Comment by t00f** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "component": "trireme-lib",
    "pr-id": "499",
    "commit-sha": "e3fa8aed6c16195973d5dfbb57119420a6c23901"
  },
  {
    "component": "enforcerd",
    "pr-id": "532",
    "commit-sha": "9e33f845175c7f11bce9ff0a9f2ca483761bcb55"
  },
  {
    "component": "gaia",
    "pr-id": "121",
    "commit-sha": "c38b8870d6afbd3cefb13766bbb89a0b2ccfad30"
  }
]
```

**Comment by t00f** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "component": "trireme-lib",
    "pr-id": "499",
    "commit-sha": "a9309dde0c71b29efab3697d8bb6bb4a272af28e"
  },
  {
    "component": "enforcerd",
    "pr-id": "532",
    "commit-sha": "096a5ed1b1496aa6caeaccb3aee177d4c45c6a9a"
  },
  {
    "component": "gaia",
    "pr-id": "121",
    "commit-sha": "fbeeed0b16005cf6bf19e7fe5ffa2dcfc254c65a"
  }
]
```

**Pull request merged**